packages/components/src/combobox-control/index.tsx

```diff
@@ -234,16 +240,17 @@ function ComboboxControl( props: ComboboxControlProps ) {
 
 	const onBlur = () => {
 		setInputHasFocus( false );
+
+		if ( ! currentOption ) {
+			onFilterValueChange( '' );
+		}
```

Comment:
If we restrict the `onFilterValueChange` reset to when there is no current option, this raises the problem of the visual input value and the consumer's filter value going out of sync.

The problem now dovetails with what you're suggesting in #65256. That behavior is not wrong or bad, and ideally our APIs would be flexible enough to support that behavior. But can we say that it is universally better or more correct over the current behavior on trunk? For example, a lot (most?) users who click on an already value-selected ComboboxControl are going to want to change that value to something else entirely. Having to delete the current input value manually is an added step. 

So my impression at this point is that we should wait until the version 2 rewrite to [Ariakit](https://ariakit.org/components/combobox), so consumers can customize behavior in a much more modular way. Otherwise, with the WP ecosystem constraints we're working within, it's hard to justify shipping a potentially breaking behavioral change unless it can be clearly established as a bug fix.

Is there anything else we can do to help support your use case now, without major changes like in #65256?

Reply:
I see what you're saying, and I agree entirely.

>Is there anything else we can do to help support your use case now

My use case is I don't want to refetch the options whenever the Combobox Control gets focused. It should only refetch the options on textinput change, otherwise I want to use whatever we have. I'm using `onFilterValueChange` to detect changes to the input so that I can fetch the options from the API.

If there's another way to do it, that's fine by me. I thought of checking if the input is empty and then not fetching anything, but then there's no way to reset the options because there's no way to differentiate an organic backspace vs. the input cleared by our API. That is, unless we add a new flag to `onFilterValueChange`, which might not be too bad.

Reply:
If it's just about reducing unnecessary API calls, can you handle it with a simple caching layer on your end? Cache the result of whatever API you call for an empty string, i.e. the unfiltered list of options.

Or is it not really about the API calls, and you actually want to show the previously filtered list of options even though the input field is visually empty?

Reply:
>Or is it not really about the API calls, and you actually want to show the previously filtered list of options even though the input field is visually empty?

It's this case. I'm not too concerned about the API calls themselves, but I want to preserve the list of options.

Reply:
And that would be an acceptable UX even without the changes in #65256? I'd think you need a way to signal that there is a filter value, and provide a way to clear it.

I'm saying this because adding more arguments to `onFilterValueChange` is a non-breaking change we could potentially make, but #65256 is not.

Reply:
I'm starting to think about that flag more.

Let's start by noting that the `inputValue` will be cleared every time the input is blurred.

If the input gets blurred and no option is selected, clicking the input again will render an empty textbox. Therefore, it makes no sense to keep the previous options because the input was closed, and no option was selected.

If the input gets blurred and there is a selected option, clicking the input again will still render an empty textbox. But if we introduce the flag, only the subset of filtered options will be presented on focus (because the callback was not called/called with a flag to ignore the change in the filter), despite the textbox itself displaying no previous filter.

So, I'm starting to think these changes don't make sense. Initially, I felt that the Combobox should respect the existing value of the filter whenever opening the suggestion list, much like a SelectControl, but I no longer think that's the case.

Reply:
Agreed. I'm afraid we'll have to wait for ComboboxControlV2 to safely customize this kind of behavior.

> That means we'll miss the tests, but I don't think it's a big deal because they were asserting use cases that don't make sense 😄

FWIW I think we can keep the "calls onFilterValueChange whenever the textbox changes" and "clears the textbox value if there is no selected value on blur" tests, with minor changes so it asserts the actual behavior on trunk.